src/editor.c

```diff
@@ -4758,24 +4758,32 @@ on_editor_scroll_event(GtkWidget *widget, GdkEventScroll *event, gpointer user_d
 {
 	GeanyEditor *editor = user_data;
 
-	/* we only handle up and down, leave the rest to Scintilla */
+	/* Scintilla handles direct left/right scrolling */
 	if (event->direction != GDK_SCROLL_UP && event->direction != GDK_SCROLL_DOWN)
 		return FALSE;
 
-	/* Handle scroll events if Alt is pressed and scroll whole pages instead of a
-	 * few lines only, maybe this could/should be done in Scintilla directly */
 	if (event->state & GDK_MOD1_MASK)
 	{
-		sci_send_command(editor->sci, (event->direction == GDK_SCROLL_DOWN) ? SCI_PAGEDOWN : SCI_PAGEUP);
+		/* scroll whole pages when Alt is pressed */
+		gint amount = (event->direction == GDK_SCROLL_DOWN) ? SCI_PAGEDOWN : SCI_PAGEUP;
+		sci_send_command(editor->sci, amount);
 		return TRUE;
 	}
 	else if (event->state & GDK_SHIFT_MASK)
 	{
+		/* translate scrolling to left/right when Shift is pressed */
 		gint amount = (event->direction == GDK_SCROLL_DOWN) ? 8 : -8;
-
 		sci_scroll_columns(editor->sci, amount);
 		return TRUE;
 	}
+	else if (!event->state || (editor_prefs.scrollwheel_zoom_disable && (event->state & GDK_CONTROL_MASK)))
```

Comment:
This doesn't actually do what you think it does (in some situations).  `event->state` contains all modifier states, including e.g. <kbd>NumLock</kbd> and others, so in the non-disabled case it'll fallback to Scintilla's handling if e.g. <kbd>NumLock</kbd> is on, and your code if it's not.
The proper solution is to filter through `keybindings_get_modifiers()`:

```c
GdkModifierType modifiers = keybindings_get_modifiers(event->state);

if (modifiers & GDK_MOD1_MASK)
    …
else if (modifiers & …
```

Reply:
Some existing code will need to be revised then.

Reply:
Possibly, did you see some? This very function was fine because it only checked whether some modifiers were present, not that they were specific -- and the ones checked weren't in the special cases for macos -- but I can totally imagine we have issues some other places (although I'd think they'd be discovered by now -- your case was a lot more subtle to see given that if your test failed the behavior was still very similar)

Reply:
I didn't understand the purpose earlier.  It makes more sense now that I see it's to work around how Macs interpret some keys.

Reply:
If I want to remove the control mask, would either of the following work for mac?
```
event->state = event->state & !GDK_CONTROL_MASK;
```

```
GdkModifierType modifiers = keybindings_get_modifiers(event->state);
event->state = modifiers & !GDK_CONTROL_MASK;
```

Reply:
It's also to filter out uninteresting modifiers like NumLock.

For removing, you could try `event->state &= ~GDK_CONTROL_MASK;`, or if inside the `if (event->state & GDK_CONTOL_MASK)`, just `event->state ^= GDK_CONTROL_MASK;`.